### ISSUE TITLE
Proposal to use Koseven name in classes

### ISSUE THREAD

**Issue by neo22s** [koseven/koseven#282]
For example, instead of class Kohana_Core => Koseven_Core.

Also replace Kohana folders for Koseven.

We could keep the Kohana name so migration it's easy and backwards compatible, although doing a find replace it's not complex.

I know does not add any functionality but I think by releasing 4.0.0 maybe it's worth to use the new branding.

Of course always we will keep credit to all the kohana team work.

What's your opinion?

**Comment by jstrobel** [comment]
Think that too. With release 4 we can rename it.

**Comment by piotrbaczek** [comment]
I'm against it. It's a lot of very tedious work and we get nothing in return. I would rather spend that time developing or improving software, writing unittest event to get us from pity 35% to a 100%.

**Comment by neo22s** [comment]
@piotrbaczek of course this has low priority, but at some point will need to be done, it's super strange and confusing for anyone that wants to start a new project using Koseven, that needs to type Kohana :S

If we release 4.0.0 it's because we also want new people to start using the project I think no?

**Comment by neo22s** [comment]
We can just simply extend the kohana class to have another one called koseven... and ready, but having all folders called kohana it's ugly as hell.

We are plenty of people, if we organize our effort properly we can make it.

I know I have not been there for almost a year but seems that now after January I will be more free (december holidays xD).

I am not the best developer ever I can handle myself, but I have been successfully leading teams and small companies with almost 0 budget for years, also an open source project which is what I make a living from.

If we all go to the same path together we can achive anything we want.

I believe that Koseven it's a really powerful framework as valid as any other one out there and we can improve that.

Also I am aware of some tensions, probably the way I write. I like to be direct and honest and I love to listen to everyone. If you knew me in person you may put a tone to my writing. It's the bad part of remote working, you do not get to know the people fully.

Thanks to all of you again.

**Comment by uBizzy** [comment]
I agree with @piotrbaczek but I fully understand @neo22s 's vision. If we ever release a Koseven v4 for sure this would be a big improvement but not the only one.

**Comment by toitzi** [comment]
This may require time...but i got some "spare time ;)". If somone just could explain to me how you want this to be done (maybe with a small example)..I can do it

**Comment by svenbw** [comment]
@toitzi That's the idea; it would be even better if this could be "scripted", if a local repo of user uses also the Kohana classes the code can be updated automatically.
I wrote some code which renames the directories, for classes I assume that some kind of reflection / token parsing is required to cover everything.

**Comment by toitzi** [comment]
Just to clearify:

A Bash Script, and when you call it everything in your "application" folder gets "rewritten" to use Koseven? ....then why not just use "Refactoring" in your IDE to accomplish the same but without the work of creating a script?

**Comment by timhj** [comment]
It's a large amount of work to go through all this and will destroy any backwards compatibility with the existing Kohana apps, any of the old modules will become way more of a hassle to get running in projects.

While a bunch of work has gone into the project since 3.2, koseven is still the Kohana frameworl but for PHP7... Changing the classnames and core will kill any chance of a resurgence of the framework and limit ongoing use only to teams with enough Devs to warrant a full review and refactor of their apps to move to the new naming conventions.

It's a great thing you're working on and the Kohana framework should continue, changing all the core classes over to Koseven will mean it's just you guys working on it forever. People would be crazy to look at the framework over laravel then.

Keep it as is, it's a lot of work with no tangible benefit to the framework and who knows whether it'll all be done before PHP 8 rolls around ;)

**Comment by svenbw** [comment]
> A Bash Script, and when you call it everything in your "application" folder gets "rewritten" to use Koseven? ....then why not just use "Refactoring" in your IDE to accomplish the same but without the work of creating a script?

Not everybody has an IDE that supports refactoring.

**Comment by toitzi** [comment]
@timhj 
Why will it destroy backwards compatibility if we extend "Koseven Classes" to keep the "Kohana" class names?

Ok i think i got that part wrong...thought koseven is a PHP7 Version of Kohana BUT there to help migrating kohana users to koseven, and - when "done" - keep going on as "independent" framework.
If Koseven was just there to "resurgence" Kohana, i don't get why renaming it to koseven in first place.

**Comment by neo22s** [comment]
@puchkovk I think that could work... like similar to jQuery has. An optional module that you can enable in case you need to use any old module.

Also agree that most modules are not PHP7 compatible.

**Comment by toitzi** [comment]
@puchkovk 

I really like the idea of doing it as a "kohana module"

if @neo22s @svenbw @jstrobel are okay with that i would rewrite it to "koseven" and write a "kohana module" for backwards compatibility and then do a PR?

**Comment by neo22s** [comment]
@toitzi also please do not forget the docs of the module I know lot of work if you need a hand let us know.

**Comment by timhj** [comment]
@neo22s - As long as the change is backwards compatible then it wouldn't be killing off potential community members. If a change gets deployed that's not backwards compatible and apps stop working (until apps do a find/replace for Kohana_ to Koseven_ across the apps entire codebase and relevant testing), it's not great. 

A huge part of the success the web's larger frameworks/languages like Wordpress, jquery and even PHP (to a large extent) has been keeping backwards compatibility between versions. New versions add new functionality and new ways of doing things but the old scripts (minus some deprecations etc.) continue to work. 

Tools that have ignored backwards compatibility have signed their own death sentences imho.. eg. the drupal 8 rewrite (technically necessary and probably makes it a better product but kills the community) [uptake and usage over time has dwindled since launch](https://imgur.com/a/rCXlxw3).

I'm just struggling with what the real benefit to the framework is of a namechange... if it goes ahead and it's backwards compatible with apps already built then all good - just for the love of good make it backwards compatible.  :)

**Comment by timhj** [comment]
@neo22s - Ah ok, less of a thing then but doing it again wouldn't be a great idea. Particularly for lurkers like me with little dev time. Merges from master that breaks apps is mean.

**Comment by toitzi** [comment]
Did Pull Reques #285 
Please check and give some feedback!

**Comment by WinterSilence** [comment]
@neo22s why not `Ko7` or `K7`? Also we can leave prefix `Kohana` unchanged, but set version as '7.0.0'.

**Comment by neo22s** [comment]
@WinterSilence yes I like too K7. Short and to the point. We where keeping the style used as kohana and was not KO. Maybe we can create a shortcut for this but call it koseven.

**Comment by WinterSilence** [comment]
@neo22s at first time we can add aliases `K7_*` for `Kohana_*` classes in autoloader.
